extra_data/reader.py

```diff
@@ -1440,16 +1441,22 @@ def open_run(proposal, run, data='raw', include='*'):
     include: str
         Wildcard string to filter data files.
     """
-    if isinstance(proposal, int):
-        proposal = 'p{:06d}'.format(proposal)
-    elif ('/' not in proposal) and not proposal.startswith('p'):
-        proposal = 'p' + proposal.rjust(6, '0')
+    if isinstance(proposal, str):
+        if ('/' not in proposal) and not proposal.startswith('p'):
+            proposal = 'p' + proposal.rjust(6, '0')
+    else:
+        # Allow integers, including numpy integers
+        proposal = 'p{:06d}'.format(index(proposal))
 
     prop_dir = find_proposal(proposal)
 
-    if isinstance(run, int):
-        run = 'r{:04d}'.format(run)
-    elif not run.startswith('r'):
-        run = 'r' + run.rjust(4, '0')
+    if isinstance(run, str):
```

Comment:
Can just use
`    if isinstance(run, str) and run.startswith('r'):
          run = run[1:]`
or add something here to convert the string to a int and check that it is in range.

Reply:
Joining the conditions in one `if` statement changes when the `else` section runs, though. There's several different ways to achieve the same thing, but I think this one is as good as any.

I'm not too concerned about the range check - I think the `FileNotFoundError` it triggers by default is clear enough. And the actual range of runs that exist is different for each proposal, in any case.

```
FileNotFoundError: [Errno 2] No such file or directory: '/gpfs/exfel/exp/SCS/201901/p002212/raw/r0888'
```

Reply:
In fact, I wonder if it's best to take the range check out, so the error for a run that doesn't exist is consistent.

Reply:
Okay - take it out. I only added it as your colleague suggested it.
However, be careful if someone stupidly passes the string "t024" instead of "r024" - is this handled?

Reply:
Oops, sorry for the inconsistent feedback.

Typo-ing something like `t024` will give the same kind of error, including the mistaken text as part of the path. We could catch things like that separately to make a better error message, but it's not high priority, so I'll leave it for another day.

Reply:
I just suggested discussing the check of a range because there is a little difference in the sense of errors: the run doesn't exist and it cannot exist in principle. But I don't think that it's important.